effect.cpp

```diff
@@ -59,6 +59,33 @@
 #include "workshop_type.h"
 #include "automaton_part.h"
 
+#define MODIFIER_EFFECT_BODY(Name, name_suffix, desc_suffic) \
+bool Effects::Name::applyToCreature(Creature* c, Creature* attacker) const { \
+  return false; \
+} \
+\
+string Effects::Name::getName(const ContentFactory* f) const { \
+  return effect->getName(f) + name_suffix; \
+} \
+\
+string Effects::Name::getDescription(const ContentFactory* f) const { \
+  return effect->getDescription(f) + desc_suffic; \
+}
+
+#define FILTER_EFFECT_BODY(Name, NameSuffix, DescSuffix) \
```

Comment:
Thinking about either converting MODIFIER_EFFECT_BODY's name_suffix, desc_suffix into lambdas, or removing lambdas from here. whatever you find more preferable.